client/src/App.tsx

```diff
@@ -0,0 +1,99 @@
+import { FormEvent, useEffect, useState } from "react";
+import "./App.css";
+import { Movie, deleteMovie, getMovies, saveMovie, updateMovie } from "./lib/data";
+import { MovieCard } from "./components/MovieCard";
+import "./movie.css";
+import { MovieForm } from "./components/MovieForm";
+
+function App() {
+  const [movies, setMovies] = useState<Movie[]>([]);
+
+  useEffect(() => {
+    async function fetchMovies() {
+      const fetchedMovies = await getMovies();
+      if (fetchedMovies) {
+        setMovies(fetchedMovies);
+      } else {
+        console.error("error fetching movies");
+      }
+    }
+    fetchMovies();
+  }, []);
+
+  async function handleAddMovie(e: FormEvent<HTMLFormElement>) {
+    e.preventDefault();
+    const form = e.currentTarget;
+    const data = new FormData(form);
+    const movie: Movie = {
```

Comment:
The easiest way to get this object is to use `Object.fromEntries(form);`

Since one of your values is a number, you can do that conversion after extracting the object.